base/ryu/Ryu.jl

```diff
@@ -17,6 +17,15 @@ neededdigits(::Type{Float64}) = 309 + 17
 neededdigits(::Type{Float32}) = 39 + 9 + 2
 neededdigits(::Type{Float16}) = 9 + 5 + 9
 
+"""
+    Ryu.neededfloatdigits(T)
+
+Number of digits necessary to represent type `T` in shortest precision.
+"""
+neededfloatdigits(::Type{Float64}) = 23
```

Comment:
"If an IEEE 754 double-precision number is converted to a decimal string with at least 17 significant digits, and then converted back to double-precision representation, the final result must match the original number"

17 + decimal point + sign bit + "e" + second sign bit + max exponent (3) = 24?

https://en.wikipedia.org/wiki/Double-precision_floating-point_format